srml/contract/src/lib.rs

```diff
@@ -105,6 +105,9 @@ pub trait Trait: fees::Trait + balances::Trait + timestamp::Trait {
 	/// A function type to get the contract address given the creator.
 	type DetermineContractAddress: ContractAddressFor<CodeHash<Self>, Self::AccountId>;
 
+	/// Fee charge.
+	type ChargeFee: ChargeFee<Self::AccountId, Amount=Self::Balance>;
```

Comment:
If we are ok with changes here regarding contracts, we need to document the requirements of `ChargeFee` implementation. I.e. `refund_fee` should fail and etc.